pymc/logprob/order.py

```diff
@@ -0,0 +1,124 @@
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+#   MIT License
+#
+#   Copyright (c) 2021-2022 aesara-devs
+#
+#   Permission is hereby granted, free of charge, to any person obtaining a copy
+#   of this software and associated documentation files (the "Software"), to deal
+#   in the Software without restriction, including without limitation the rights
+#   to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+#   copies of the Software, and to permit persons to whom the Software is
+#   furnished to do so, subject to the following conditions:
+#
+#   The above copyright notice and this permission notice shall be included in all
+#   copies or substantial portions of the Software.
+#
+#   THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+#   IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+#   FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+#   AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+#   LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+#   OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+#   SOFTWARE.
+
+from typing import List, Optional
+
+import pytensor.tensor as pt
+
+from pytensor.graph.basic import Node
+from pytensor.graph.fg import FunctionGraph
+from pytensor.graph.rewriting.basic import node_rewriter
+from pytensor.tensor.math import Max
+from pytensor.tensor.random.op import RandomVariable
+from pytensor.tensor.var import TensorVariable
+
+from pymc.logprob.abstract import (
+    MeasurableVariable,
+    _logcdf_helper,
+    _logprob,
+    _logprob_helper,
+)
+from pymc.logprob.rewriting import measurable_ir_rewrites_db
+
+
+class MeasurableMax(Max):
+    """A placeholder used to specify a log-likelihood for a max sub-graph."""
+
+
+MeasurableVariable.register(MeasurableMax)
+
+
+@node_rewriter([Max])
+def find_measurable_max(fgraph: FunctionGraph, node: Node) -> Optional[List[TensorVariable]]:
+    rv_map_feature = getattr(fgraph, "preserve_rv_mappings", None)
+    if rv_map_feature is None:
+        return None  # pragma: no cover
+
+    if isinstance(node.op, MeasurableMax):
+        return None  # pragma: no cover
+
+    base_var = node.inputs[0]
+
+    if base_var.owner is None:
+        return None
+
+    # NonRVS must be rejected
+    if not isinstance(base_var.owner.op, RandomVariable):
+        return None
+
+    # univariate i.i.d. test which also rules out other distributions
+    if isinstance(base_var.owner.op, RandomVariable):
```

Comment:
perhaps just `else` here? The condition is identical to the line above. The comment can go inside the `else` statement:
```python
if not isinstance(base_var.owner.op, RandomVariable):
        # Non-RVS must be rejected
        return None
else:
       # univariate i.i.d. test which also rules out other distributions
       for params in base_var.owner.inputs[3:]:
            if params.type.ndim != 0:
                return None

Reply:
Or just remove the second condition altogether since returning `None` would exit this function:
```python
if not isinstance(base_var.owner.op, RandomVariable):
        # Non-RVS must be rejected
        return None

 # univariate i.i.d. test which also rules out other distributions
 for params in base_var.owner.inputs[3:]:
      if params.type.ndim != 0:
          return None
```
This now becomes a Python style question...